core/src/main/scala/zio/config/ReadFunctions.scala

```diff
@@ -2,129 +2,260 @@ package zio.config
+                       .either
+                       .map({
+                         case Left(error) =>
+                           singleton(errors(error): ConfResult[Vector[K], ::[B]])
+                         case Right(opt) =>
+                           opt match {
+                             case Some(values) =>
+                               mapCons(withIndex(values.value)) {
+                                 case (Some(values), _) =>
+                                   seqEitherCons(
+                                     mapCons(values)(value => propertyType.read(value))
+                                   ).fold(
+                                     r =>
+                                       errors(
+                                         ReadError.ParseError(
+                                           paths :+ path,
+                                           ReadFunctions.parseErrorMessage(r.value.toString, r.typeInfo)
+                                         )
+                                       ),
+                                     e => Exists[Vector[K], ::[B]](paths :+ path, e)
+                                   )
+                                 case (None, id) =>
+                                   Errors(ReadError.MissingValue[Vector[K]](paths :+ path, Some(id)))
+                               }
+
+                             case None =>
+                               singleton(
+                                 errors(ReadError.MissingValue[Vector[K]](paths :+ path, None)): ConfResult[
+                                   Vector[K],
+                                   ::[B]
+                                 ]
+                               )
+                           }
+                       })
           } yield result
 
         case s: Sequence[K, V1, B] @unchecked =>
           val Sequence(config) = s
-          loop(config, paths).map(list => singleton(list))
+          loop(config, paths).map(list => {
+            singleton(seqConfResult(list))
+              .asInstanceOf[::[ConfResult[Vector[K], ::[B]]]] // Required only for scala 2.12
+          })
 
         case ConfigDescriptor.Nested(path, c) =>
           loop(c, paths :+ path)
 
         case cd: ConfigDescriptor.XmapEither[K, V1, a, B] =>
           val ConfigDescriptor.XmapEither(c, f, _) = cd
-          loop(c, paths).flatMap { as =>
-            foreach(as)(a => {
+          loop(c, paths).flatMap { results =>
+            foreach(results)(a => {
               ZIO
-                .fromEither(f(a))
-                .bimap(
-                  err =>
-                    singleton(
-                      ReadError.fatalError[Vector[K], V1](paths, new RuntimeException(err))
-                    ),
-                  res => res
+                .succeed(
+                  a.flatMap(
+                    values =>
+                      seqEitherCons(mapCons(values)(f)) match {
```

Comment:
This can be improved actually. Collect the errors here as well.

In that way 

   [
       {  
          k : [v1, v2, v3]
       }   
    
       
       {  
          k : null
       }   

   ]
The error when parsing each element in first k, and the error when trying to fetch the second k should all be accumulated